src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5886,63 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            _ = EatToken(SyntaxKind.AsteriskToken);
+
+            switch (CurrentToken.ContextualKind)
+            {
+                case SyntaxKind.CdeclKeyword:
+                case SyntaxKind.ManagedKeyword:
+                case SyntaxKind.StdcallKeyword:
+                case SyntaxKind.ThiscallKeyword:
+                case SyntaxKind.UnmanagedKeyword:
+                    _ = EatContextualToken(CurrentToken.ContextualKind);
+                    break;
+
+                default:
+                    if (CurrentToken.Kind != SyntaxKind.LessThanToken)
+                    {
+                        _ = EatToken();
+                    }
+                    break;
+            }
+
+            _ = EatToken(SyntaxKind.LessThanToken);
+
+            do
+            {
+                switch (CurrentToken.Kind)
+                {
+                    case SyntaxKind.ReadOnlyKeyword:
+                        _ = EatToken(SyntaxKind.ReadOnlyKeyword);
+                        _ = EatToken(SyntaxKind.RefKeyword);
+                        break;
+
+                    case SyntaxKind.RefKeyword:
+                    case SyntaxKind.InKeyword:
+                    case SyntaxKind.OutKeyword:
+                        _ = EatToken(CurrentToken.Kind);
+                        break;
+                }
+
+                ScanType(out _);
```

Comment:
1. doc that you don't need these values.
2. does scantype not return something?  if it does, do you want to do `_ =` for consistency?

Reply:
For 1, all the `ScanType` functions are pretty consistent here in that they only need the last token. If you open the file in the IDE, for example, you'll see fading from the dead assignment analyzer. We should probably replace a lot of things with `_ =` for clarity.

---
In reply to: [348016168](https://github.com/dotnet/roslyn/pull/39879#discussion_r348016168) [](ancestors = 348016168)